### PR TITLE
[MANUAL MIRROR] Fixes toilet bongs, and fixes a serious bug in the crafting system

### LINKED ISSUE #18732: Crafting Menu unable to craft Toilet bong and possible other items.

### PR DESCRIPTION
**Pull request by vinylspiders**
## ORIGINAL PR: https://github.com/tgstation/tgstation/pull/72893

Merge this PR first: https://github.com/Skyrat-SS13/Skyrat-tg/pull/18919.

Fixes #18732

## Changelog

:cl:
fix: toilet bongs crafting recipes
fix: fixed bug with crafting that led to unexpected behavior; e.g. recipe requirements not being consumed, machinery being consumed when they shouldn't be
refactor: cleaned up some old code in the recipes file, added support for structures in recipes
/:cl:

### CODE REVIEW

**Comment by nevimer** [comment]
For manual mirrors, cherry pick the tg commit to have no issues and retain the metadata.

**Comment by nevimer** [comment]
Additionally, credit the tg author in the changelog. (You are the og author so it doesn't matter here)

**Comment by vinylspiders** [comment]
> For manual mirrors, cherry pick the tg commit to have no issues and retain the metadata.

Ah I thought that's what I did. Did I cherry pick the wrong commit? 

I did `git cherry-pick <commit>` with commit being the one that got merged.

**Comment by nevimer** [comment]
> > For manual mirrors, cherry pick the tg commit to have no issues and retain the metadata.
> 
> Ah I thought that's what I did. Did I cherry pick the wrong commit? 
> 
> I did `git cherry-pick <commit>` with commit being the one that got merged.

Disregard, I saw that you were both the author and commiter on the commit and came to the conclusion of  a mishap before I noticed that it was accurate to upstream

**Comment by GoldenAlpharex** [comment]
> > For manual mirrors, cherry pick the tg commit to have no issues and retain the metadata.
> 
> Ah I thought that's what I did. Did I cherry pick the wrong commit?
> 
> I did `git cherry-pick <commit>` with commit being the one that got merged.

What I usually do it `git cherry-pick -m 1 <commit hash>`, I don't know the difference but it usually preserves the author, in my case.

**Comment by vinylspiders** [comment]
> > > For manual mirrors, cherry pick the tg commit to have no issues and retain the metadata.
> > 
> > 
> > Ah I thought that's what I did. Did I cherry pick the wrong commit?
> > I did `git cherry-pick <commit>` with commit being the one that got merged.
> 
> Disregard, I saw that you were both the author and commiter on the commit and came to the conclusion of a mishap before I noticed that it was accurate to upstream

Oh whew, I was worried for a second.

**Comment by GoldenAlpharex** [comment]
I'd recommend just re-doing this now that the dependency was merged, or you can fix the conflicts yourself, up to you.

**Comment by vinylspiders** [comment]
> I'd recommend just re-doing this now that the dependency was merged, or you can fix the conflicts yourself, up to you.

Yeah. In hindsight redoing it would probably have been faster. Fixed the conflicts though, should be good now.

File: tgui/packages/tgui/interfaces/PersonalCrafting.tsx [hunk 1]
```diff
@@ -648,6 +649,11 @@ const RecipeContentCompact = ({ item, craftable, busy, mode }, context) => {
                 {item.machinery &&
                   ', ' +
                     item.machinery
+                      .map((item) => data.atom_data[(item as any) - 1]?.name)
+                      .join(', ')}
+				{item.structures &&
```
Comment (GoldenAlpharex) [inline tgui/packages/tgui/interfaces/PersonalCrafting.tsx]:
```suggestion
                {item.structures &&
```

File: tgui/packages/tgui/interfaces/PersonalCrafting.tsx [hunk 2]
```diff
@@ -779,6 +785,14 @@ const RecipeContent = ({ item, craftable, busy, mode, diet }, context) => {
                       <AtomContent key={atom_id} atom_id={atom_id} amount={1} />
                     ))}
                   </Box>
+                )}
+				 {item.structures && (
```
Comment (GoldenAlpharex) [inline tgui/packages/tgui/interfaces/PersonalCrafting.tsx]:
```suggestion
                {item.structures && (
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
These two will cause linting problems I'm pretty sure, related to a merge skew that happened yesterday.

**Comment by GoldenAlpharex** [comment]
The suggestion seems to be weird, basically make sure that it's got the same code as in <https://github.com/tgstation/tgstation/pull/72908/files#> for the personal crafting file.

**Comment by nevimer** [comment]
some times reapplying the cherry pick on top again and can resolve issues too. I'm not certain of your scenario, but it tends to work for me.

**Comment by vinylspiders** [comment]
> The suggestion seems to be weird, basically make sure that it's got the same code as in https://github.com/tgstation/tgstation/pull/72908/files# for the personal crafting file.

Makes sense to me! done.

**Comment by vinylspiders** [comment]
> some times reapplying the cherry pick on top again and can resolve issues too. I'm not certain of your scenario, but it tends to work for me.

I'll try that next time, thanks for the suggestion.

**Review by GoldenAlpharex — APPROVED** [review]
Good, thank you!

**Pull request merged**